### PR TITLE
feat(overlays): provide option to cancel 'willClose' and 'willOpen' events

### LINKED ISSUE #2118: Feature Request: introduce method to abort closing a modal
### Feature Description

Swap content from a modal instead of closing it. To do so, there has to be the option prevent the dialog from closing after clicking on the close button (X) 

Example: If you want to edit a traveler when registering travelers and then click close, the entire modal should not be closed but you should jump to the previous view

More information: https://flow.sbb.ch/browse/SBBCH-16518 

### Definition of Done

- [ ] Component is implemented
- [ ] Documentation is provided
- [ ] Tests are implemented
- [ ] E2E Tests are implemented
- [ ] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by TomMenga**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #2118 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- `will-open` and `will-close` events are now cancellable

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/autocomplete/autocomplete.ts [hunk 1]
```diff
@@ -123,9 +123,10 @@ export class SbbAutocomplete extends LitElement {
       return;
     }
 
-    this._state = 'opening';
-    this._willOpen.emit();
-    this._setOverlayPosition();
+    if (this._willOpen.emit()) {
+      this._state = 'opening';
+      this._setOverlayPosition();
+    }
```
Comment (jeripeierSBB) [inline src/components/autocomplete/autocomplete.ts:129]: This comment would be for all files (not only this one):
I personally would prefer exits with "return" in those cases, as then the flow (especially if there is more content in a block) is not nested. But probably you have a different opinion?

```suggestion
    if (!this._willOpen.emit()) {
      return
    }
    this._state = 'opening';
    this._setOverlayPosition();
```

**Review by jeripeierSBB — COMMENTED** [review]
I especially like all the added tests :)

File: src/components/autocomplete/autocomplete.ts (hunk 1 above)
Comment (TomMenga) [reply src/components/autocomplete/autocomplete.ts:129]: I don't have a strong preference between the two solutions. I can change it 😄

**Review by jeripeierSBB — APPROVED** [review]
LGTM, please wait with merge until we have the first stable release of lit

**Pull request merged**